miasm2/ir/ir.py

```diff
@@ -173,21 +173,20 @@ def merge_multi_affect(self, affect_list):
                 continue
 
             # Find collision
-            e_colision = reduce(
-                lambda x, y: x + y, [e.get_modified_slice() for e in expr_list])
-
+            e_colision = reduce(lambda x, y: set(x).union(y),
+                                (e.get_modified_slice() for e in expr_list))
```

Comment:
By bad, it's better written as this:

```Python
e_colision = reduce(lambda x, y: x.union(y), (e.get_modified_slice() for e in expr_list), set())
```

This is a tiny fix but I think it's better as it avoids to call the `set()` constructor on a `set` object for each element from the generator. Sorry I didn't write it like that in the first place.

The rest of the patch looks good to me.